pallets/gear/src/manager/journal.rs

```diff
@@ -314,11 +314,16 @@ where
                     dispatch.is_reply(),
                 ),
                 (None, None) => Pallet::<T>::split(message_id, dispatch.id(), dispatch.is_reply()),
-                (Some(_gas_limit), Some(_reservation_id)) => {
-                    // TODO: #1828
-                    unreachable!(
-                        "Sending dispatch with gas limit from reservation \
-                    is currently unimplemented and there is no way to send such dispatch"
+                (Some(gas_limit), Some(reservation_id)) => {
+                    Pallet::<T>::split_with_value(
+                        reservation_id,
+                        dispatch.id(),
+                        gas_limit,
+                        dispatch.is_reply(),
+                    );
+                    Pallet::<T>::remove_gas_reservation_with_task(
+                        dispatch.source(),
+                        reservation_id,
```

Comment:
I don't see the test for the correctness of what's here.
If you call `Pallet::<T>::remove_gas_reservation_with_task` it will consume reservation node and return the gas back to the origin. So the amount of gas for the `dispatch` will be the one, which was provided within `split_from_value`. Is it a required business logic? Because it seems that instead of merging gas from reservation and the "split", we use only "split" and free the reservation.

Reply:
@ark0f @breathx @gshep 

Please take a look!!!

Reply:
if you do only `split_with_value`, then reservation node won't be used.

Reply:
hmmm, so mb need a new method